### PR TITLE
MM-41963 Gekidou Post Options Delete and Edit

### PR DESCRIPTION
**Pull request by avinashlng1080**
#### Summary
This PR includes the EDIT and DELETE option from the Post Option menu.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-41963
https://mattermost.atlassian.net/browse/MM-41964

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: app/actions/remote/post.ts [hunk 1]
```diff
@@ -552,3 +552,27 @@ export const togglePinPost = async (serverUrl: string, postId: string) => {
...
+
+    let client: Client;
+    try {
+        client = NetworkManager.getClient(serverUrl);
+    } catch (error) {
+        return {error};
+    }
+
+    try {
+        client.deletePost(postId);
```
Comment (larkox) [inline app/actions/remote/post.ts:570]: Should we handle any error on the client function? Or we expect anything that may go wrong to throw an exception? If so, should we at least await it?

File: app/screens/post_options/options/delete_post_option.tsx [hunk 2]
```diff
@@ -1,22 +1,53 @@
...
+            [{
+                text: formatMessage({id: 'mobile.post.cancel', defaultMessage: 'Cancel'}),
+                style: 'cancel',
+                onPress: () => dismissBottomSheet(Screens.POST_OPTIONS),
+            }, {
+                text: formatMessage({id: 'post_info.del', defaultMessage: 'Delete'}),
+                style: 'destructive',
+                onPress: () => {
+                    deletePost(serverUrl, postId);
+                    dismissBottomSheet(Screens.POST_OPTIONS);
```
Comment (larkox) [inline app/screens/post_options/options/delete_post_option.tsx:39]: Should we dismiss the bottom sheet before the alert? (0/5, more a UX question)

File: app/actions/remote/post.ts (hunk 1 above)
Comment (enahum) [reply app/actions/remote/post.ts:570]: yup definitely await and do nothing if fails

File: app/screens/post_options/options/delete_post_option.tsx (hunk 2 above)
Comment (enahum) [reply app/screens/post_options/options/delete_post_option.tsx:39]: yes, the way this works is:
1. User press on the delete option
2. Alert is displayed
3. User presses on `Delete`
4. the delete action is called and the bottom sheet closes.

File: app/screens/post_options/options/edit_option.tsx [hunk 3]
```diff
@@ -1,22 +1,30 @@
...
-        //todo:
-    };
+type Props = {
+    post: PostModel;
+}
+const EditOption = ({post}: Props) => {
+    const onPress = useCallback(() => {
+        // https://mattermost.atlassian.net/browse/MM-41991
+        goToScreen('EDIT_SCREEN_NOT_IMPLEMENTED_YET', '', {post});
+        dismissBottomSheet(Screens.POST_OPTIONS);
```
Comment (enahum) [inline app/screens/post_options/options/edit_option.tsx:20]:
```suggestion
        await dismissBottomSheet(Screens.POST_OPTIONS);
        goToScreen('EDIT_SCREEN_NOT_IMPLEMENTED_YET', '', {post});
```

File: app/screens/post_options/options/edit_option.tsx [hunk 4]
```diff
@@ -1,22 +1,30 @@
...
 
-const EditOption = () => {
-    const handleEdit = () => {
-        //todo:
-    };
+type Props = {
+    post: PostModel;
+}
+const EditOption = ({post}: Props) => {
+    const onPress = useCallback(() => {
```
Comment (enahum) [inline app/screens/post_options/options/edit_option.tsx:17]:
```suggestion
    const onPress = useCallback(async () => {
```

**Review by enahum — CHANGES_REQUESTED** [review]

File: app/screens/post_options/options/delete_post_option.tsx [hunk 5]
```diff
@@ -1,22 +1,53 @@
...
+        Alert.alert(
+            formatMessage({id: 'mobile.post.delete_title', defaultMessage: 'Delete Post'}),
+            formatMessage({
+                id: 'mobile.post.delete_question',
+                defaultMessage: 'Are you sure you want to delete this post?',
+            }),
+            [{
+                text: formatMessage({id: 'mobile.post.cancel', defaultMessage: 'Cancel'}),
+                style: 'cancel',
+                onPress: () => dismissBottomSheet(Screens.POST_OPTIONS),
```
[the remark this answers is not in the document]
Comment (avinashlng1080) [reply app/screens/post_options/options/delete_post_option.tsx:33]: removed

**Review by enahum — APPROVED** [review]

**Review by larkox — APPROVED** [review]

**Pull request merged**